projects/plugins/crm/tests/php/entities/class-transaction-entity-test.php

```diff
@@ -0,0 +1,87 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+
+namespace Automattic\Jetpack\CRM\Entities\Tests;
+
+use Automattic\Jetpack\CRM\Entities\Factories\Transaction_Factory;
+use Automattic\Jetpack\CRM\Entities\Transaction;
+use Automattic\Jetpack\CRM\Tests\JPCRM_Base_Integration_Test_Case;
+
+/**
+ * Test Event Manager system.
+ *
+ * @covers Automattic\Jetpack\CRM\Entities
+ */
+class Transaction_Entity_Test extends JPCRM_Base_Integration_Test_Case {
+
+	/**
+	 * @testdox Test that transaction entity is created from input data.
+	 */
+	public function test_transaction_entity_from_input_data() {
+
+		$transaction_data = $this->generate_transaction_data();
+
+		// Create the Transaction instance from the transaction data (tidy DAL format)
+		$transaction = Transaction_Factory::create( $transaction_data );
+
+		$this->assertInstanceOf( Transaction::class, $transaction );
+
+		// Check that the transaction data field values are the same to the Transaction instance field values.
+		foreach ( $transaction_data as $key => $value ) {
+			if ( ! property_exists( $transaction, $key ) ) {
+				continue;
+			}
+			$this->assertEquals( $value, $transaction->$key, "Transaction property $key does not match" );
+		}
+	}
+
+	/**
+	 * @testdox Test create transaction entity from input data and insert in DB via DAL.
+	 */
+	public function test_create_transaction_from_input_data_and_insert_into_DB() {
+
+		$transaction_data = $this->generate_transaction_data();
+
+		// Create the Transaction instance from the transaction data (tidy DAL format)
```

Comment:
The factories are temporary code until we have repository classes for all entities, so we shouldn't put too much focus on them now, but inline comments should end with a `.` 😄